src/System.Text.Json/tests/Serialization/ExceptionTests.cs

```diff
@@ -79,5 +78,211 @@ public static void ThrownFromReader()
                 Assert.Equal(8, inner.BytePositionInLine);
             }
         }
+
+        [Fact]
+        public static void PathForDictionaryFails()
+        {
+            try
+            {
+                JsonSerializer.Parse<Dictionary<string, int>>(@"{""Key1"":1, ""Key2"":bad}");
```

Comment:
What happens if the JSON is `@"{""Key1"":1, ""Key2"`

Reply:
What happens if the JSON is @"{""Key1"":1, ""Key2"""

Reply:
Since there is no `:` delimiter, it would be `$.Key1`

I'll add a test.

Reply:
IMO the path should be `$`. The error didn't happen at `$.Key1`. Key1 just happened to be the last property read. The error happened on attempting to read a the next name/value from `$`.

Reply:
Situations like this is why I kept things simple in Json.NET and only included the known parent segments.

Reply:
The current Path semantics follow the reader semantics. If we want different semantics we'd likely have to change the reader to not accept a space for the delimiter cc @ahsonkhan

Reply:
An error didn't happen parsing `$.Key1`. Key1 is done. It's finished and in the past. The state should have been popped off the stack when its value was finished parsing. The new error happened when parsing a parameter on the root object: `$`. I don't know what the reader semantics have to do with it.

Reading through the JSON, and what the path would be if an error was thrown:

Read: `{`
Path: `$'

Read: `Key1`
Path: `$` (because you haven't finished reading Key1)

Read: `1`
Path: `$.Key1`

Read: `Key2`
Path: `$` (Key1 was popped off the stack once its value was read. Now you start reading the next property on the root object)

Reply:
The serializer currently does not track state separate from the reader. So if the reader says a property name or array element is "read" successfully then the serializer treats that as part of the path. Let's discuss in next meeting.